src/AsyncTCP.cpp

```diff
@@ -1017,9 +1036,12 @@ size_t AsyncClient::write(const char* data) {
 
 size_t AsyncClient::write(const char* data, size_t size, uint8_t apiflags) {
     size_t will_send = add(data, size, apiflags);
-    if(!will_send || !send()) {
+    if (!will_send) {
         return 0;
     }
+    while (connected() && !send()) {
+        taskYIELD();
+    }
```

Comment:
@TienHuyIoT: when using SSE events in ESPAsycnWebServer, I think this busy loop can wait forever in case of a WIFi disconnection because the client is still seen there and connected